frontend/test/lit/test_device_api.py

```diff
@@ -119,3 +120,23 @@ def circuit_split():
 
 
 test_preprocess()
+
+
+def test_simple_circuit_set_shots():
+    """Test that a circuit with the new device API is compiling to MLIR."""
+    dev = qml.device("lightning.qubit", wires=2)
+
+    @qjit(target="mlir")
+    @partial(qml.set_shots, shots=2048)
+    @qml.qnode(device=dev)
+    def circuit():
+        qml.Hadamard(wires=0)
+        qml.CNOT(wires=[0, 1])
+        return qml.expval(qml.PauliZ(wires=0))
+
+    # CHECK: [[shots:%.+]] = arith.constant 2048 : i64
+    # CHECK: quantum.device shots([[shots]]) {{.*}}
+    print(circuit.mlir)
+
+
+test_simple_circuit_set_shots()
```

Comment:
Is this a device API test? 

Even if there's only the one test in it, it might make sense to just have a `frontend/test/lit/test_set_shots.py` file or something like that.

Reply:
https://github.com/PennyLaneAI/catalyst/pull/1784/commits/b4ba5980eb58b01a48bb61a2200dbe88f6788c2f